### ISSUE TITLE
Hello, can you give me some advice?

### ISSUE THREAD

**Issue by dj1367250510** [CNJianLiu/MH6D#1]
When I ran your training code, I encountered the following problem. This problem seemed to point to torch/optim/lr_scheduler.py. I tried to open the source code to view it, but I could not solve the problem. Because it seems that the source code is written as self.step().

![F4CS%0UT{O8MP8DP6Q(4VDG](https://github.com/CNJianLiu/MH6D/assets/74551357/9661854a-a7e8-4009-b6a9-6a0624aaeb7e)

**Comment by CNJianLiu** [comment]
Hi, thank you for being interested in our work. I tried to run our code again and did not find this problem. Maybe you can use "        self.lr_scheduler = optim.lr_scheduler_CyclicLR(self.optimizer, base_lr=1e-5, max_lr=1e-3, step_size_up= cfg.max_epoch * cfg.num_mini_batch_per_epoch // 6, mode='triangular')" replace the original code.

**Comment by dj1367250510** [comment]
> Hi, thank you for being interested in our work. I tried to run our code again and did not find this problem. Maybe you can use " self.lr_scheduler = optim.lr_scheduler_CyclicLR(self.optimizer, base_lr=1e-5, max_lr=1e-3, step_size_up= cfg.max_epoch * cfg.num_mini_batch_per_epoch // 6, mode='triangular')" replace the original code.

Thank you for your previous answers, I have some questions to confirm with you.
1. According to the experimental results of your article, training without REAL-WORLD data, I think it is the training and verification mentioned in your code. I would like to confirm whether the REAL W LABEL / REAL W/O LABELL in the table below correspond to the test code respectively supervisedwithReal275Mask or supervisedwoReal275Mask set in
2. When I finished training train.py, I thought it corresponded to the data results of training with REAL-WORLD data in your table, and I also did a test, but found that the results were very different. I wanted to know where I did it. wrong.
![image](https://github.com/CNJianLiu/MH6D/assets/74551357/6a33dab6-de7e-47d1-9ae7-5224490acc1e)

**Comment by CNJianLiu** [comment]
Hi, when training with REAL275, you should directly use the model already trained on CAMERA25, which is explained in the Implementation Details section of the paper. We use the supervisedwithReal275Mask segmentation results for testing, you can also use the supervisedwoReal275Mask, and the results will not differ greatly.

**Comment by dj1367250510** [comment]
> Hi, when training with REAL275, you should directly use the model already trained on CAMERA25, which is explained in the Implementation Details section of the paper. We use the supervisedwithReal275Mask segmentation results for testing, you can also use the supervisedwoReal275Mask, and the results will not differ greatly.

Hello, do you mean that I should use the model30 trained by train_sys.py before, and train to 40epoch with train.py according to the max_epoch set in 3MH_sysreael.yaml?

**Comment by CNJianLiu** [comment]
Yes

**Comment by dj1367250510** [comment]
> Yes

Hello, according to your previous suggestion, I did the corresponding operation. I used the model30 weight trained on the previous camera data set, continued training on train.py, and got the model40 weight. I used the model40 weight for testing. The result It doesn't come close to the results in your article, so I'd like some advice.

![image](https://github.com/CNJianLiu/MH6D/assets/74551357/ca5e4d21-7fe1-4e5d-a9d3-1493bbdd2b89)

![image](https://github.com/CNJianLiu/MH6D/assets/74551357/35a7a4ac-9084-4aa6-9056-9907c02e4f87)

![image](https://github.com/CNJianLiu/MH6D/assets/74551357/55a31e8f-024c-4bfc-8539-ed19342bc5d4)

**Comment by CNJianLiu** [comment]
Are you sure you use the same segmentation results as [DPDN](https://github.com/JiehongLin/Self-DPDN)? How about using the model_30 to test directly?

**Comment by dj1367250510** [comment]
> 是否确定使用与 [DPDN](https://github.com/JiehongLin/Self-DPDN) 相同的分割结果？使用model_30直接测试怎么样？

I confirmed that I was using the same segmentation results as DPDN, because I have reproduced SELF-DPDN before, and all the data sets I used were inherited from the previous reproduction. The results measured directly using model30 are worse than those measured using model40. At present, the test work I have done in training without real-world data is close to the results in your paper, but in training with real-world data, the test work I have done is always very different.

**Comment by CNJianLiu** [comment]
You mean use our released model_30? I just evaluate again and got the same results.

**Comment by dj1367250510** [comment]
> You mean use our released model_30? I just evaluate again and got the same results.

In fact, I did not use the model30 you gave in this code. I trained trainsys.py to model30 myself and did the testing.The first time I tried to use the model30 file you provided, I encountered some environment reading problems that I couldn't solve, so I ran trainsys.py myself for training and testing.

**Comment by CNJianLiu** [comment]
Can you check your training_logger? When saving the model every 10 epochs, the learning rate should be close to the minimum. I think you may have a problem [here](https://github.com/CNJianLiu/MH6D/blob/45c1bce889edbda56e2187cd3e806e2f30d1d88f/utils/solver.py#L38). When training with CAMERA25 only, it is '//6', and when training with REAL275, it should be '//8'.

**Comment by dj1367250510** [comment]
> Can you check your training_logger? When saving the model every 10 epochs, the learning rate should be close to the minimum. I think you may have a problem [here](https://github.com/CNJianLiu/MH6D/blob/45c1bce889edbda56e2187cd3e806e2f30d1d88f/utils/solver.py#L38). When training with CAMERA25 only, it is '//6', and when training with REAL275, it should be '//8'.

Hello, can I add your contact information to communicate with you?My qq is 1367250510

**Comment by luxilandu** [comment]
> > Can you check your training_logger? When saving the model every 10 epochs, the learning rate should be close to the minimum. I think you may have a problem [here](https://github.com/CNJianLiu/MH6D/blob/45c1bce889edbda56e2187cd3e806e2f30d1d88f/utils/solver.py#L38). When training with CAMERA25 only, it is '//6', and when training with REAL275, it should be '//8'.
> 
> Hello, can I add your contact information to communicate with you?My qq is 1367250510

How is it going? Have you reproduced the results in the paper?
